sirepo/auth/email.py

```diff
@@ -147,6 +152,13 @@ def _send_smtp(user_data, uri):
         m = self.auth_db.model(UserModel)
         u = m.unchecked_search_by(unverified_email=email)
         if not u:
+            # POSIT: Email has been validated by _parse_email so simple split is ok
+            if email.split("@")[1] in _cfg.deny_access_domains:
```

Comment:
Make the conditional check a function that returns the email: `n.new(unverified_email=_assert_allow(email))`. 

(My pedantic self would say that deny_access_domains should be checked for empty, but since this is a low cost split, I will only comment parenthetically. :)

Reply:
34cc6ad44